amethyst_utils/src/app_root_dir.rs

```diff
@@ -1,21 +1,28 @@
 //! Provides the directory of the executable.
 
-use std::{env, fs};
+use std::{env, io, path};
 
 /// Returns the cargo manifest directory when running the executable with cargo
 /// or the directory in which the executable resides otherwise,
 /// traversing symlinks if necessary.
-pub fn application_root_dir() -> String {
-    env::var("CARGO_MANIFEST_DIR").unwrap_or_else(|_| {
-        let mut path = env::current_exe().expect("Failed to find executable path.");
-        while let Ok(target) = fs::read_link(path.clone()) {
-            path = target;
-        }
-        String::from(
-            path.parent()
-                .expect("Failed to get parent directory of the executable.")
-                .to_str()
-                .expect("Failed to get valid string from path to the executable."),
-        )
-    })
+pub fn application_root_dir() -> Result<path::PathBuf, io::Error> {
+    if let Some(manifest_dir) = env::var_os("CARGO_MANIFEST_DIR") {
+        let manifest_dir = manifest_dir.into_string().map_err(|_| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                "`CARGO_MANFIEST_DIR` exists, but contains non-utf8 data",
```

Comment:
That should not be a requirement though.

Reply:
I re-implemented the old behavior, but I agree that it shouldn't be a requirement.